src/pages/billing/Project.js

```diff
@@ -531,34 +532,40 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
         !!displayName && span({ style: { flexShrink: 0, fontWeight: 600, fontSize: 14, margin: '0 0.75rem 0 0' } }, 'Billing Account:'),
-        !!displayName && span({ style: { flexShrink: 0 } }, displayName),
-        isOwner && h(Link, {
-          tooltip: 'Change Billing Account',
+        !!displayName && span({ style: { flexShrink: 0, marginRight: '0.5rem' } }, displayName),
+        isOwner && h(MenuTrigger, {
+          closeOnClick: true,
+          side: 'bottom',
           style: { marginLeft: '0.5rem' },
-          onClick: async () => {
-            if (Auth.hasBillingScope()) {
-              setShowBillingModal(true)
-            } else {
-              await authorizeAndLoadAccounts()
-              setShowBillingModal(Auth.hasBillingScope())
-            }
-          }
-        }, [icon('edit', { size: 12 })]),
-        isOwner && h(Link, {
-          tooltip: 'Remove Billing Account',
-          style: { marginLeft: '0.5rem' },
-          // (CA-1586) For some reason the api sometimes returns string null, and sometimes returns no field, and sometimes returns null. This is just to be complete.
-          disabled: billingProject.billingAccount === 'null' || billingProject.billingAccount === undefined || billingProject.billingAccount === null,
-          baseColor: colors.danger,
-          onClick: async () => {
-            if (Auth.hasBillingScope()) {
-              setShowBillingRemovalModal(true)
-            } else {
-              await authorizeAndLoadAccounts()
-              setShowBillingRemovalModal(Auth.hasBillingScope())
-            }
-          }
-        }, [icon('times-circle', { size: 12 })]),
+          content: h(Fragment, [
+            h(MenuButton, {
+              onClick: async () => {
+                if (Auth.hasBillingScope()) {
+                  setShowBillingModal(true)
+                } else {
+                  await authorizeAndLoadAccounts()
+                  setShowBillingModal(Auth.hasBillingScope())
+                }
+              }
+            }, ['Change Billing Account']),
+            h(MenuButton, {
+              onClick: async () => {
+                if (Auth.hasBillingScope()) {
+                  setShowBillingRemovalModal(true)
+                } else {
+                  await authorizeAndLoadAccounts()
+                  setShowBillingRemovalModal(Auth.hasBillingScope())
+                }
+              },
+              // (CA-1586) For some reason the api sometimes returns string null, and sometimes returns no field, and sometimes returns null. This is just to be complete.
+              disabled: billingProject.billingAccount === 'null' || _.isNil(billingProject.billingAccount)
+            }, ['Remove Billing Account'])
+          ])
+        }, [
+          h(Link, { 'aria-label': 'Billing account menu', style: { display: 'flex', alignItems: 'center' } }, [
+            icon('cardMenuIcon', { size: 16 })
```

Comment:
I think you may want to add `'aria-haspopup': 'menu'`.

Reply:
Added in a69db71945c9285793a055a48f1cff00711640c3